src/track/serato/tags.cpp

```diff
@@ -152,49 +165,94 @@ double SeratoTags::guessTimingOffsetMillis(
     // used to determine which case the MP3 is classified in. See the following
     // PR for more detailed information:
     // https://github.com/mixxxdj/mixxx/pull/2119
-
     double timingOffset = 0;
     if (taglib::getFileTypeFromFileName(filePath) == taglib::FileType::MP3) {
+        const QString primaryDecoderName =
+                getPrimaryDecoderNameForFilePath(filePath);
+        // There should always be an MP3 decoder available
+        VERIFY_OR_DEBUG_ASSERT(!primaryDecoderName.isEmpty()) {
+            return 0;
+        }
+        bool timingOffsetGuessed = false;
 #if defined(__COREAUDIO__)
-        Q_UNUSED(signalInfo);
-
-        int timingShiftCase = mp3guessenc_timing_shift_case(filePath.toStdString().c_str());
-
-        // TODO: Find missing timing offsets
-        switch (timingShiftCase) {
-        case EXIT_CODE_CASE_A:
-            timingOffset = -12;
-            break;
-        case EXIT_CODE_CASE_B:
-            timingOffset = -40;
-            break;
-        case EXIT_CODE_CASE_C:
-        case EXIT_CODE_CASE_D:
-            timingOffset = -60;
-            break;
+        if (primaryDecoderName == mixxx::SoundSourceProviderCoreAudio::kDisplayName) {
+            DEBUG_ASSERT(!timingOffsetGuessed);
+            DEBUG_ASSERT(timingOffset == 0);
+
+            Q_UNUSED(signalInfo)
+            int timingShiftCase = mp3guessenc_timing_shift_case(filePath.toStdString().c_str());
+
+            // TODO: Find missing timing offsets
+            switch (timingShiftCase) {
+            case EXIT_CODE_CASE_A:
+                timingOffset = -12;
+                break;
+            case EXIT_CODE_CASE_B:
+                timingOffset = -40;
+                break;
+            case EXIT_CODE_CASE_C:
+            case EXIT_CODE_CASE_D:
+                timingOffset = -60;
+                break;
+            }
+
+            timingOffsetGuessed = true;
         }
-#elif defined(__MAD__) || defined(__FFMPEG__)
-        switch (signalInfo.getSampleRate()) {
-        case 48000:
-            timingOffset = -24;
-            break;
-        case 44100:
-            // This is an estimate and tracks will vary unpredictably within ~1 ms
-            timingOffset = -26;
-            break;
-        default:
-            qWarning()
-                    << "Unknown timing offset for Serato tags with sample rate"
-                    << signalInfo.getSampleRate();
+#endif
+#if defined(__MAD__) || defined(__FFMPEG__)
+        bool usingMadOrFFmpeg = false;
+#endif
```

Comment:
this endif can be removed, let's just nest these to make it easier to grasp and save some lines.